scipy/linalg/_basic.py

```diff
@@ -254,8 +243,24 @@ def solve(a, b, lower=False, overwrite_a=False,
     else:
         trans = 0
         norm = '1'
-
-    anorm = _matrix_norm(norm, a1, check_finite)
+    
+    # Currently we do not have the other forms of the norm calculators
+    #   lansy, lanpo, lanhe.
+    # However, in any case they only reduce computations slightly...
+    if assume_a == 'diagonal':
+        anorm = _matrix_norm_diagonal(a1, check_finite)
+    elif assume_a == 'tridiagonal':
+        anorm = _matrix_norm_tridiagonal(norm, a1, check_finite)
+    elif assume_a == 'banded':
+        n_below, n_above = bandwidth(a1) if n_below is None else (n_below, n_above)
+        a2, n_below, n_above = ((a1.T, n_above, n_below) if transposed
```

Comment:
Huh, so these were reversed below, but it wasn't caught be the tests. Could you take a look at the test and see if they could be made a little stronger so they would have caught this? (Is `a1` symmetric? I can't imagine we don't ever provide `banded` as the structure.)

Reply:
Ah I actually got confused for the same reason you first wrote but I now see that the original order was correct too. Do you want me to put them back the other way around?

Reply:
No, I think we both found the original order confusing, so let's keep this.